                                                                                                                      syntax = "proto3";

package Trevor;

message HealthCheckRequest {
  string service_name = 1;
  string uid = 2;
}

message HealthCheckResponse {
  string service_name = 1;
  string uid = 2;
  string status = 3;  // "healthy", "disconnected", "shutting_down"
}

// Portfolio Management Messages
message PortfolioRequest {
  string account_id = 1;
  string requester_uid = 2;
  repeated string symbols = 3;  // Optional: specific symbols to include
}

message PortfolioResponse {
  string account_id = 1;
  double total_value = 2;
  double cash_balance = 3;
  string status = 4;  // "active", "suspended", "closed"
  repeated Position positions = 5;
}

message Position {
  string symbol = 1;
  double quantity = 2;
  double average_cost = 3;
  double current_price = 4;
  double market_value = 5;
  double unrealized_pnl = 6;
}

// Market Data Messages
message MarketDataUpdate {
  string symbol = 1;
  double price = 2;
  int64 volume = 3;
  int64 timestamp = 4;
  string exchange = 5;
}

// Trading Messages
message TradeRequest {
  string account_id = 1;
  string symbol = 2;
  string side = 3;  // "buy", "sell"
  double quantity = 4;
  string order_type = 5;  // "market", "limit", "stop"
  double price = 6;  // For limit/stop orders
  string requester_uid = 7;
}

message TradeResponse {
  string account_id = 1;
  string order_id = 2;
  string status = 3;  // "filled", "partial", "rejected", "pending"
  double filled_quantity = 4;
  double average_fill_price = 5;
  string reason = 6;  // For rejections
}
